source/PlayerInfo.cpp

```diff
@@ -2171,6 +2194,70 @@ void PlayerInfo::SetTravelDestination(const Planet *planet)
 
 
 
+void PlayerInfo::QueueRelocation(const Planet *destination, bool flagshipOnly)
+{
+	relocation = Relocation(destination, flagshipOnly);
+}
+
+
+
+void PlayerInfo::DoQueuedRelocation()
+{
+	if(!relocation.relocationPlanet || !FlagshipPtr())
+		return;
+	if(!planet)
+	{
+		relocation.relocationPlanet = nullptr;
+		return;
+	}
+
+	flagship->SetSystem(relocation.relocationPlanet->GetSystem());
+	flagship->SetPlanet(relocation.relocationPlanet);
+	if(!relocation.relocateFlagshipOnly)
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsParked() && !ship->IsDestroyed() && ship->GetPlanet() == planet)
+			{
+				ship->SetSystem(relocation.relocationPlanet->GetSystem());
+				ship->SetPlanet(relocation.relocationPlanet);
+			}
+	system = relocation.relocationPlanet->GetSystem();
+	planet = relocation.relocationPlanet;
```

Comment:
We're moving the player's ship, but we're not moving the player's cargo. When you're landed, the cargo from all your ships gets moved to the pooled cargo hold in PlayerInfo. By only moving the ships, we're leaving the cargo behind.

We should either move all the cargo, or if we're only moving the flagship, move an amount of cargo that fits on your flagship.

Reply:
ok, thanks

Reply:
moved one step from takeoff to leaveplanet that should address this issue